api/proto/teleport/legacy/types/types.proto

```diff
@@ -2856,7 +2856,6 @@ message RoleOptions {
   // DeviceTrustMode is the device authorization mode used for the resources
   // associated with the role.
   // See DeviceTrust.Mode.
-  // Reserved for future use, not yet used by Teleport.
```

Comment:
FYI @hugoShaka, the new proto check on master and v16 is not running correctly. This PR passed on master and v16, but failed on v15 and 14.

- https://github.com/gravitational/teleport/actions/runs/9897707845/job/27342848403?pr=44097#step:20:1
- https://github.com/gravitational/teleport/actions/runs/9897708390/job/27342850975?pr=44098#step:18:1